src/main/java/com/google/jenkins/plugins/k8sengine/KubernetesEnginePublisher.java

```diff
@@ -220,10 +260,36 @@ public FormValidation doCheckManifestPattern(@QueryParameter String value) {
       if (Strings.isNullOrEmpty(value)) {
         return FormValidation.error(Messages.KubernetesEnginePublisher_ManifestRequired());
       }
-
       return FormValidation.ok();
     }
 
+    public ListBoxModel doFillZoneItems(
+        @AncestorInPath Jenkins context,
+        @QueryParameter("projectId") final String projectId,
+        @QueryParameter("credentialsId") final String credentialsId) {
+      ListBoxModel items = new ListBoxModel();
+      items.add("- none -", "");
+      if (!Strings.isNullOrEmpty(projectId) && !Strings.isNullOrEmpty(credentialsId)) {
```

Comment:
For improved readability, returning early should be preferred over nesting. Suggest re-factoring to negate the codition and return an empty list.

Reply:
Done